### PR TITLE
MES 4188 uncouple recouple cat be component

### PR DESCRIPTION
**Pull request by Cullima**
## Description
- New uncouple recouple component
- Added component to Cat B+E test Report
- Updated translations displayed for vehicle faults on Debrief Page
- Unit tests and analytics will be submitted as separate PRs

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [x] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/assets/i18n/cy.json [hunk 1]
```diff
@@ -68,7 +68,8 @@
       "forwardParkObservation": "Parcio Ymlaen - gwyliadwriaeth",
       "vehicleChecks": "Gwirio’r cerbyd",
       "controlledStop": "Stopio dan Reolaeth",
-      "eyesightTest": "Prawf Golwg"
+      "eyesightTest": "Prawf Golwg",
+      "uncoupleRecouple": "Uncouple / Adennill"
```
Comment (rossfauldsbjss) [inline src/assets/i18n/cy.json:72]: Is the welsh for Uncouple  uncouple? or does it just not exist in the welsh language?

**Review by rossfauldsbjss — APPROVED** [review]
looks good

File: src/pages/test-report/components/uncouple-recouple/uncouple-recouple.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<ion-row>
+  <ion-col col-20 no-padding>
+    <competency-button class="uncouple-recouple-tick"
+                       [onTap]="toggleUncoupleRecouple"
```
Comment (szabikr) [inline src/pages/test-report/components/uncouple-recouple/uncouple-recouple.html]: This formatting is not aligned with other html files. Just use 2 spaces please.

File: src/pages/test-report/components/uncouple-recouple/uncouple-recouple.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+      if (!this.isSeriousMode && !this.isDangerousMode && this.faultCount() > 0) {
+        return true;
+      }
+      return false;
+    }
+    return !(this.hasDangerousFault() || this.hasSeriousFault() || this.faultCount() > 0);
+  }
+
+  toggleUncoupleRecouple = (): void => {
```
Comment (szabikr) [inline src/pages/test-report/components/uncouple-recouple/uncouple-recouple.ts:128]: Any unit tests for these functions?

File: src/assets/i18n/cy.json (hunk 1 above)
Comment (Cullima) [reply src/assets/i18n/cy.json:72]: I am reliably (?) informed it is the same in Welsh and English. Matt tells me all translations will be professionally checked at a later date.

**Review by szabikr — APPROVED** [review]

File: src/pages/test-report/components/uncouple-recouple/uncouple-recouple.ts (hunk 3 above)
Comment (Cullima) [reply src/pages/test-report/components/uncouple-recouple/uncouple-recouple.ts:128]: will be added in a separate PR (MES-4189)

File: src/pages/test-report/components/uncouple-recouple/uncouple-recouple.html (hunk 2 above)
Comment (Cullima) [reply src/pages/test-report/components/uncouple-recouple/uncouple-recouple.html]: done

**Pull request merged**